lib/bap/bap_configuration.ml

```diff
@@ -0,0 +1,829 @@
+     compiled with $(b,bapbuild) tool using $(b,bapbuild PATH.plugin)
+     command." in
+    Arg.(value @@ opt_all string [] @@ info ["l"] ~docv:"PATH" ~doc)
+
+  let load_path =
+    let doc =
+      "Add $(docv) to a set of search paths. Plugins found in the
+    search paths will be loaded automatically." in
+    Arg.(value @@ opt_all string []
+         @@ info ["L"; "load-path"] ~docv:"PATH" ~doc)
+
+  let list_plugins =
+    let doc = "List available plugins" in
+    Arg.(value @@ flag @@ info ["list-plugins"] ~doc)
+
+  let disable_plugin =
+    let doc = "Don't load $(docv) automatically" in
+    Arg.(value @@ opt_all string []
+         @@ info ["disable-plugin"] ~docv:"PLUGIN" ~doc)
+
+  let no_auto_load =
+    let doc = "Disable auto loading of plugins" in
+    Arg.(value @@ flag @@ info ["disable-autoload"] ~doc)
+
+  let global_term =
+    let combine _ _ _ _ _ = () in
+    Term.(const combine
+          $ load
+          $ load_path
+          $ list_plugins
+          $ disable_plugin
+          $ no_auto_load)
+
+  exception Plugin_not_found of string
+
+  let open_plugin_exn name =
+    let name = name ^ ".plugin" in
+    if Sys.file_exists name then Plugin.of_path name
+    else raise (Plugin_not_found name)
+
+  let open_plugin ~verbose name =
+    try
+      let p = open_plugin_exn name in
+      if verbose then
+        eprintf "Loader: opened plugin %s\n" @@ Plugin.name p;
+      Some p
+    with
+    | Plugin_not_found name ->
+      eprintf "Loader: can't find plugin %s\n" name;
+      None
+    | exn when verbose ->
+      eprintf "Loader: can't open plugin %s: %a\n"
+        name Exn.pp exn;
+      None
+    | exn ->
+      eprintf "Loader: can't open plugin %s\n" name; None
+
+  (* We will not ignore errors on plugins loaded explicitly
+      by a user with `-l` option. *)
+  let load_plugin p = ok_exn (Plugin.load p)
+
+
+  let autoload_plugins ~library ~verbose ~exclude =
+    Plugins.run ~library ~exclude ()
+      ~don't_setup_handlers:true
+  (* we don't want to fail the whole platform if some
+     plugin has failed, we will just emit an error message.  *)
+
+
+  let get_opt ~default argv opt  =
+    Option.value (fst (Term.eval_peek_opts ~argv opt)) ~default
+
+  let excluded argv =
+    let module Pat = String.Search_pattern in
+    let pat = Pat.create "--no-" in
+    Array.fold ~init:[] argv ~f:(fun plugins opt ->
+        match Pat.index pat opt with
+        | Some 0 -> Pat.replace_first pat opt "" :: plugins
+        | _ -> plugins)
+
+  let print_plugins_and_exit excluded plugins =
+    List.iter plugins ~f:(fun p ->
+        let status = if List.mem excluded (Plugin.name p)
+          then "[-]" else "[+]" in
+        printf "  %s %-16s %s@." status (Plugin.name p) (Plugin.desc p));
+    exit 0
+
+  let verbose =
+    let doc = "Print verbose output to log file" in
+    Arg.(value @@ flag @@ info ["verbose"] ~doc)
+
+  type action = Keep | Drop
+
+  let filter_option options prefix =
```

Comment:
Currently, the `filter_option` function exists to get rid of `--no-PLUGIN` type of arguments. If we remove it, then we'll need to add in `--no-taint`, `--no-propagate-taint`, `--no-ida`, etc (i.e. for every plugin). While this would be an easy change to make, it'd cause further bloat to the man page. Are we sure we want to do this? If so, then I can make that change, and remove this completely.